src/main/java/net/minecraftforge/event/entity/player/ItemFishedEvent.java

```diff
@@ -0,0 +1,86 @@
+/*
+ * Minecraft Forge
+ * Copyright (c) 2016.
+ *
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation version 2.1
+ * of the License.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ */
+
+package net.minecraftforge.event.entity.player;
+
+import com.google.common.base.Preconditions;
+import net.minecraft.entity.projectile.EntityFishHook;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.NonNullList;
+
+import javax.annotation.Nonnegative;
+import java.util.List;
+
+/**
+ * This event is called when a player fishes an item.
+ * You can change the items the player will get using {@link #getItemStacks()}
+ * You can also change the damage the rod will take using {@link #setRodDamage(int)}
+ * You can use {@link #getHookEntity()} to get stuff based on the entity, like {@link EntityFishHook#fishApproachAngle}
+ *
+ * This event is not {@link net.minecraftforge.fml.common.eventhandler.Cancelable}
+ */
+public class ItemFishedEvent extends PlayerEvent
+{
+    private final List<ItemStack> stacks = NonNullList.create();
+    private final EntityFishHook hook;
+    private int rodDamage;
+
+    public ItemFishedEvent(List<ItemStack> stacks, int rodDamage, EntityFishHook hook)
+    {
+        super(hook.getAngler());
+        this.stacks.addAll(stacks);
+        this.rodDamage = rodDamage;
+        this.hook = hook;
+    }
+
+    /**
+     * Get the damage the rod will take
+     * @return The damage the rod will take
+     */
+    public int getRodDamage()
+    {
+        return rodDamage;
+    }
+
+    /**
+     * Set the damage the fishing rod will take
+     * @param rodDamage The damage the rod will take. Must be greater or equals zero
+     */
+    public void setRodDamage(@Nonnegative int rodDamage)
```

Comment:
I think this would be better named as `damageRodBy`. I feel like using `RodDamage` doesn't provide meaningful context for which this method is used.

```java
event.damageRodBy(10);
```

is much more clear in its intent than

```java
event.setRodDamage(10);
```

in my opinion. Changing this would also require a change in the docs:

```java
/**
 * Specifies the amount of damage that the fishing rod should take. This is not added to the pre-existing damage to be taken.
 * @param rodDamage Must be greater than or equal to zero.
 */
```